### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Removes all deprecations marked for removal in 24.1.0.

Entrypoints removal is postponed given ongoing discussion (https://github.com/conda/conda-build/issues/5081).

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Review by jezdez — DISMISSED** [review]

File: conda_build/cli/main_convert.py [hunk 1]
```diff
@@ -35,8 +37,8 @@
 """
 
 
-def parse_args(args):
-    p = ArgumentParser(
+def parse_args(args: Iterable[str] | None) -> tuple[ArgumentParser, Namespace]:
```
Comment (mbargull) [inline conda_build/cli/main_convert.py:40]: In the other modules you used `args: Sequence[str]` for `parse_args`/`execute`.
`argparse.parse_args` should be fine with either (`argparse` does a `args = list(args)` internally).
Just wanted to point out the slight inconsistency

File: news/5151-remove-deprecations
Comment (jezdez) [inline news/5151-remove-deprecations:1]: Since #5149 is already merged, let's add this to CHANGELOG.md.

**Review by jezdez — CHANGES_REQUESTED** [review]

File: conda_build/cli/main_convert.py (hunk 1 above)
Comment (jezdez) [reply conda_build/cli/main_convert.py:40]: @mbargull Since we're in a rush to get this out, could you work on a patch for this?

**Review by mbargull — DISMISSED** [review]
LGTM!

File: conda_build/cli/main_convert.py (hunk 1 above)
Comment (jezdez) [reply conda_build/cli/main_convert.py:40]: Nevermind, I'll do it

Comment (mbargull) [reply conda_build/cli/main_convert.py]: > Nevermind, I'll do it

Ah, didn't see that message.
I just pushed a change to use `Sequence[str]` here too.
Feel free to amend how you see fit.

Comment (jezdez) [reply conda_build/cli/main_convert.py]: zing!

Comment (jezdez) [reply conda_build/cli/main_convert.py]: This change is exactly I was about to push, no harm done :)

Comment (jezdez) [reply conda_build/cli/main_convert.py]: Thank yoU!

**Review by jezdez — APPROVED** [review]

**Review by beeankha — APPROVED** [review]

**Pull request merged**